File: Src/OfficeAutomation.API/OfficeAutomation.API/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using OfficeAutomation.Core.DTOs;
using OfficeAutomation.Services.Interfaces;
using System;
using System.Threading.Tasks;

namespace OfficeAutomation.API.Controllers
{
    // ✅ Defines this class as an API controller
    [ApiController]

    // ✅ Sets the base route for this controller: /api/Auth
    [Route("api/[controller]")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        // ==========================
        // Constructor
        // ==========================
        // Injects the authentication service
        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }

        // ==========================
        // POST: /api/Auth/login
        // ==========================
        // Handles user login requests
        [HttpPost("login")]
        public async Task<IActionResult> Login(
            [FromBody] LoginDto login,          // User login credentials
            [FromQuery] string captchaToken,    // Captcha token for validation
            [FromQuery] string clientType,      // Client type (web, mobile, etc.)
            [FromQuery] string ipAddress,       // User IP address
            [FromQuery] string deviceInfo       // User device information
        )
        {
            try
            {
                // Call AuthService to perform login
                var result = await _authService.LoginAsync(
                    login.UserName,
                    login.Password,
                    captchaToken,
                    clientType,
                    ipAddress,
                    deviceInfo
                );

                // Return successful login response with tokens
                return Ok(result);
            }
            catch (Exception ex)
            {
                // Return error response if login fails
                return BadRequest(new { message = ex.Message });
            }
        }
    }
}

File: Fullstack-JS/Backend/officeautomation-NodeJS/src/controllers/AuthController.js
const AuthService = require("../services/AuthService");

/**
 * Controller layer for handling authentication-related HTTP requests
 * Handles request/response and delegates business logic to AuthService
 */
class AuthController {
  /**
   * POST /auth/login
   * Authenticate user with username and password
   * Returns a JWT token on success
   * @param {Object} req - Express request object
   * @param {Object} res - Express response object
   */
  static async login(req, res) {
    try {
      const { username, password } = req.body;

      // Delegate login logic to AuthService
      const result = await AuthService.login(username, password);

      // Send successful response with JWT token
      res.json({ message: "Login successful", token: result.token });
    } catch (err) {
      // Handle errors (e.g., authentication failure, server error)
      res.status(err.status || 500).json({ message: err.message || "Server error" });
    }
  }
}

module.exports = AuthController;
